pkg/arvo/lib/ethio.hoon

```diff
@@ -0,0 +1,105 @@
+::  ethio: Asynchronous Ethereum input/output functions.
+::.
+/+  stdio
+::
+|*  [out-poke-data=mold out-peer-data=mold]
+=>  |%
+    ++  stdio  (^stdio out-poke-data out-peer-data)
+    --
+|%
+::  +request-rpc: send rpc request, with retry
+::
+++  request-rpc
+  |=  [url=@ta id=(unit @t) req=request:rpc:ethereum]
+  =/  m  (async:stdio ,json)
+  ^-  form:m
+  |^  %+  (retry json)  `10
+      =/  m  (async:stdio ,(unit json))
+      ^-  form:m
+      =/  =request:http
+        :*  method=%'POST'
+            url=url
+            header-list=['Content-Type'^'application/json' ~]
+            ^=  body
+            %-  some  %-  as-octt:mimes:html
+            %-  en-json:html
+            (request-to-json:rpc:ethereum id req)
+        ==
+      ;<  ~  bind:m  (send-request:stdio request)
+      ;<  rep=(unit client-response:iris)  bind:m
+        take-maybe-response:stdio
+      ?~  rep
+        (pure:m ~)
+      (parse-response u.rep)
+  ::
+  ++  retry
+    |*  result=mold
+    |=  [crash-after=(unit @ud) computation=_*form:(async:stdio (unit result))]
+    =/  m  (async:stdio ,result)
+    =|  try=@ud
+    |^  |-  ^-  form:m
+        =*  loop  $
+        ?:  =(crash-after `try)
+          (async-fail:stdio %retry-too-many ~)
```

Comment:
I wonder whether this might be a bad idea.  We don't want to just give up, we actually want to keep trying forever, though probably after yielding to a new event.  I guess that's basically what your "set a timer regardless of what happens" thing does.

Reply:
Yeah, I think this is "safe enough" for the reason you describe. Should keep an eye out for issues with this, but don't think it's blocking.